tools/cldr-apps/src/main/java/org/unicode/cldr/web/api/VoteAPIHelper.java

```diff
@@ -372,18 +372,33 @@ private static Dashboard.ReviewNotification[] getOnePathDash(
     // separated from
     // the HTTP response concerns, so that VoteAPI and XPathAlt can share code without the
     // awkwardness of forbiddenIsOk.
-    static Response handleVote(
+    public static Response handleVote(
             String loc,
             String xp,
             String value,
             int voteLevelChanged,
             final CookieSession mySession,
             boolean forbiddenIsOk) {
+        // translate this call into jax-rs Response
+        try {
+            final VoteResponse r = getHandleVoteResponse(loc, xp, value, voteLevelChanged, mySession, forbiddenIsOk);
+            return Response.ok(r).build();
+        } catch (Throwable se) {
+            return new STError(se).build();
+        }
+    }
+    /** this function is the implementation of handleVote() but does not use any jax-rs, for unit tests */
+    public static VoteResponse getHandleVoteResponse(String loc,
+            String xp,
+            String value,
+            int voteLevelChanged,
+            final CookieSession mySession,
+            boolean forbiddenIsOk) throws SurveyException {
         VoteResponse r = new VoteResponse();
         mySession.userDidAction();
         CLDRLocale locale = CLDRLocale.getInstance(loc);
         if (!UserRegistry.userCanModifyLocale(mySession.user, locale)) {
-            return Response.status(Status.FORBIDDEN).build();
+           throw new SurveyException(ErrorCode.E_NO_PERMISSION, "Not allowed to modify " + locale);
```

Comment:
did this 3-space indent get past spotless somehow?

Reply:
fixed